scripts/setup_juno_optee.sh

```diff
@@ -0,0 +1,2484 @@
+else
+	echo " `basename $DST_KERNEL` already exist, not cloning"
+fi
+
+cd $DEV_PATH
+if [ ! -d "$DST_OPTEE_OS" ]; then
+	git clone $SRC_OPTEE_OS
+else
+	echo " `basename $DST_OPTEE_OS` already exist, not cloning"
+fi
+
+cd $DEV_PATH
+if [ ! -d "$DST_OPTEE_CLIENT" ]; then
+	git clone $SRC_OPTEE_CLIENT && cd $DST_OPTEE_CLIENT && git reset --hard $STABLE_CLIENT_COMMIT
+else
+	echo " `basename $DST_OPTEE_CLIENT` already exist, not cloning"
+fi
+
+cd $DEV_PATH
+if [ ! -d "$DST_OPTEE_LINUXDRIVER" ]; then
+	git clone $SRC_OPTEE_LINUXDRIVER && cd $DST_OPTEE_LINUXDRIVER && git reset --hard $STABLE_LINUXDRIVER_COMMIT
+else
+	echo " `basename $DST_OPTEE_LINUXDRIVER` already exist, not cloning"
+fi
+
+cd $DEV_PATH
+if [ ! -d "$DST_OPTEE_TEST" ] && [ -n "$HAVE_ACCESS_TO_OPTEE_TEST" ]; then
+	git clone $SRC_OPTEE_TEST && cd $DST_OPTEE_TEST && git reset --hard $STABLE_OPTEE_TEST_COMMIT
+else
+	echo " `basename $DST_OPTEE_TEST` already exist (or no access), not cloning"
+fi
+
+################################################################################
+# Download and install the needed toolchains                                   #
+################################################################################
+mkdir -p $DEV_PATH/toolchains
+cd $DEV_PATH/toolchains
+
+if [ ! -f "${AARCH64_NONE_GCC_VERSION}.tar.xz" ]; then
+	wget $SRC_AARCH64_NONE_GCC
+fi
+if [ ! -d "$DST_AARCH64_NONE_GCC" ]; then
+	echo "Untar $AARCH64_NONE_GCC_VERSION ..."
+	tar xf ${AARCH64_NONE_GCC_VERSION}.tar.xz && mv $AARCH64_NONE_GCC_VERSION $DST_AARCH64_NONE_GCC
+fi
+
+
+if [ ! -f "${AARCH64_GCC_VERSION}.tar.xz" ]; then
+	wget $SRC_AARCH64_GCC
+fi
+if [ ! -d "$DST_AARCH64_GCC" ]; then
+	echo "Untar $AARCH64_GCC_VERSION ..."
+	tar xf ${AARCH64_GCC_VERSION}.tar.xz && mv $AARCH64_GCC_VERSION $DST_AARCH64_GCC
+fi
+
+if [ ! -f "${AARCH32_GCC_VERSION}.tar.xz" ]; then
+	wget $SRC_AARCH32_GCC
+fi
+if [ ! -d "$DST_AARCH32_GCC" ]; then
+	echo "Untar $AARCH32_GCC_VERSION ..."
+	tar xf ${AARCH32_GCC_VERSION}.tar.xz && mv $AARCH32_GCC_VERSION $DST_AARCH32_GCC
+fi
+
+################################################################################
+# Generate the build script for Linux kernel                                   #
+################################################################################
+cd $DEV_PATH
+cat > $DEV_PATH/build_linux.sh << EOF
+#/bin/bash
+export PATH=$DST_AARCH64_NONE_GCC/bin:\$PATH
+export CROSS_COMPILE=$DST_AARCH64_NONE_GCC/bin/aarch64-none-elf-
+cd $DST_KERNEL
+
+# make ARCH=arm64 CROSS_COMPILE=aarch64-none-elf- mrproper
+# make ARCH=arm64 mrproper
+if [ ! -f ".config" ]; then
+	make ARCH=arm64 defconfig
+fi
+
+if [ ! -e arch/arm64/boot/dts/juno.dts.orig ]; then
+	patch -N -b arch/arm64/boot/dts/juno.dts < ${DEV_PATH}/juno.dts.linux-linaro-tracking.${STABLE_KERNEL_COMMIT}.patch;
+fi
+if [ ! -e .config.orig ]; then
+	patch -N -b .config < ${DEV_PATH}/config.linux-linaro-tracking.${STABLE_KERNEL_COMMIT}.patch;
+fi
```

Comment:
Can we move the patch to a separate sub-folder? Instead of having it in the root of `${DEV_PATH}`?

The .config patching worries me also.  What about Jerome's question above about make defconfig. Have we checked if there is a config dedicated for Juno? (I couldn't find any when looking at ARM-TF's kernel branch).

Also, if we temporary decide to use a patch, wouldn't it still be better to add the needed stuff to the Kconfig's and Makefiles instead and have that in the *.patch? By doing so, you could still generate the .config file using normal make commands.